### PR TITLE
sleep_ms -> sleep

### LINKED ISSUE #29621: Remove deprecated `sleep_ms` usage from TRPL, replace with `sleep` and `Duration`
`sleep_ms` is deprecated as of https://github.com/rust-lang/rust/pull/29604, update the docs to reflect this.

### PR DESCRIPTION
**Pull request by steveklabnik**
Now that thread::sleep is a real thing, let's use it

Fixes #29621

r? @bstrie

### CODE REVIEW

File: src/doc/trpl/concurrency.md [hunk 1]
```diff
@@ -131,7 +132,7 @@ fn main() {
         });
     }
 
-    thread::sleep_ms(50);
+    thread::sleep(Duration::from_millis(50);
```
Comment (aravind-pg) [inline src/doc/trpl/concurrency.md]: Needs another close paren.

File: src/doc/trpl/dining-philosophers.md [hunk 2]
```diff
@@ -614,8 +617,8 @@ We have three new lines. We’ve added an argument, `table`. We access the
 the fork at that particular index. That gives us access to the `Mutex` at that
 index, and we call `lock()` on it. If the mutex is currently being accessed by
 someone else, we’ll block until it becomes available. We have also a call to
-`thread::sleep_ms` between the moment first fork is picked and the moment the
-second forked is picked, as the process  of picking up the fork is not
+`thread::sleep` between the moment first fork is picked and the moment the
```
Comment (apasel422) [inline src/doc/trpl/dining-philosophers.md]: s/moment first/moment the first/

File: src/doc/trpl/concurrency.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/trpl/concurrency.md]: figures, with the `ignore` I didn't see it :/

**Pull request merged**